### PR TITLE
[nexus3] Add install plugins

### PR DESCRIPTION
**Pull request by kwangil-ha**
https://github.com/stevehipwell/helm-charts/issues/265

### CODE REVIEW

**Comment by stevehipwell** [comment]
@kwangil-ha could you provide some context and justification for this PR? It's usually a good idea to open an issue first before creating a large PR so the requirements can be discussed.

**Comment by stevehipwell** [comment]
@kwangil-ha you should be able to remove the chown init container as we're setting `securityContext.fsGroup` for the pod. Other than that you ned to bump the chart version (it's a minor version upgrade) and create the changelog entry for the new version as well as add the changes to the chart annotations (removing markdown formatting). Once this is done I can enable the CI to run on the PR.

**Comment by stevehipwell** [comment]
@kwangil-ha the version should be `v4.5.0` for a minor version bump, you've only done a patch bump. The `fsgroup` setting means you should be able to remove the whole chown init container as the supplemental group will be added to the user in each container.

**Comment by stevehipwell** [comment]
@kwangil-ha you shouldn't need to make any changes to the security context or have an init container to chown the filesystem; K8s will set the supplemental group for filesystem access.

**Comment by kwangil-ha** [comment]
There are already the chown init container.
https://github.com/stevehipwell/helm-charts/blob/master/charts/nexus3/templates/deployment.yaml#L55-L66
I just provided the chown init container as a shell script like sonarqube helm chart.
https://github.com/stevehipwell/helm-charts/blob/master/charts/sonarqube/templates/deployment.yaml#L36-L49
Is it ok to remove both code blocks above?

**Comment by stevehipwell** [comment]
@kwangil-ha I'll take a look in the morning and get back to you with some code suggestions for testing.

**Review by stevehipwell — COMMENTED** [review]
If you make the suggested changes and test the chart locally; the volume-mount init container might be needed to create the /etc sub directory but I don't think it should and it shouldn't need to do chown.

File: charts/nexus3/templates/commands.yaml [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+  name: {{ template "nexus3.commandsName" . }}
+  labels:
+    {{- include "nexus3.labels" . | nindent 4 }}
+data:
+  chown-volume-mounts.sh: |
+    #!/usr/bin/env sh
+    set -eu
+
+    mkdir -p /nexus-data/etc
+
```
[the remark this answers is not in the document]
Comment (kwangil-ha) [reply charts/nexus3/templates/commands.yaml:13]: What is the code below?
https://github.com/stevehipwell/helm-charts/blob/master/charts/nexus3/templates/deployment.yaml#L63

File: charts/nexus3/templates/deployment.yaml [hunk 2]
```diff
@@ -36,6 +36,20 @@ spec:
...
+          env:
+          {{- toYaml . | nindent 12 }}
+          {{- end }}
+          command: ["/opt/scripts/chown-volume-mounts.sh"]
+          volumeMounts:
+            - mountPath: /opt/scripts/chown-volume-mounts.sh
+              name: commands
+              subPath: chown-volume-mounts.sh
+            - mountPath: /nexus-data
+              name: {{ include "nexus3.pvcname" . }}
```
[the remark this answers is not in the document]
Comment (kwangil-ha) [reply charts/nexus3/templates/deployment.yaml:52]: What is the code below?
https://github.com/stevehipwell/helm-charts/blob/master/charts/nexus3/templates/deployment.yaml#L55-L66

File: charts/nexus3/templates/deployment.yaml [hunk 3]
```diff
@@ -44,26 +58,29 @@ spec:
...
-          args: ["-c", "mkdir -p /nexus-data/etc; chown -R 200:200 /nexus-data;"]
+          command: ["/opt/scripts/download-plugins.sh"]
           volumeMounts:
-            - mountPath: /nexus-data
-              name: {{ include "nexus3.pvcname" . }}
+            - mountPath: /opt/scripts/download-plugins.sh
+              name: commands
+              subPath: download-plugins.sh
+            - mountPath: /opt/sonatype/nexus/deploy
+              name: deploy
```
[the remark this answers is not in the document]
Comment (kwangil-ha) [reply charts/nexus3/templates/deployment.yaml:82]: The nexus composer plugin must be copied to the <nexus_dir>/deploy path. not /nexus-data
ref. https://github.com/sonatype-nexus-community/nexus-repository-composer#more-permanent-install

File: charts/nexus3/templates/deployment.yaml [hunk 4]
```diff
@@ -105,6 +122,10 @@ spec:
           volumeMounts:
             - mountPath: /nexus-data
               name: {{ include "nexus3.pvcname" . }}
+          {{- if .Values.installPlugins.enabled }}
+            - mountPath: /opt/sonatype/nexus/deploy
+              name: deploy
+          {{- end }}
```
[the remark this answers is not in the document]
Comment (kwangil-ha) [reply charts/nexus3/templates/deployment.yaml:128]: The nexus composer plugin must be copied to the <nexus_dir>/deploy path. not /nexus-data
ref. https://github.com/sonatype-nexus-community/nexus-repository-composer#more-permanent-install

File: charts/nexus3/templates/deployment.yaml [hunk 5]
```diff
@@ -147,6 +168,14 @@ spec:
         {{- else }}
           emptyDir: {}
         {{- end }}
+      {{- if .Values.installPlugins.enabled }}
+        - name: deploy
+          emptyDir: {}
+      {{- end }}
```
[the remark this answers is not in the document]
Comment (kwangil-ha) [reply charts/nexus3/templates/deployment.yaml:174]: The nexus composer plugin must be copied to the <nexus_dir>/deploy path. not /nexus-data
ref. https://github.com/sonatype-nexus-community/nexus-repository-composer#more-permanent-install

**Comment by kwangil-ha** [comment]
If you don't want to add installing plugins for nexus, I will close this PR
https://github.com/stevehipwell/helm-charts/issues/265

**Comment by stevehipwell** [comment]
@kwangil-ha I'll take a look at a solution for you as well as my own housekeeping on the init container.

**Pull request closed without merging**